### PR TITLE
Do not add Tidepool domain email to Mailchimp

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate

### CODE REVIEW

**Comment by darinkrauss** [comment]
[<XS> Prevent tidepool.org accounts from being added to MailChimp mailing lists.](https://trello.com/c/Q48QkFXt/294-xs-prevent-tidepoolorg-accounts-from-being-added-to-mailchimp-mailing-lists)

File: user/mailchimp/mailchimp_test.go [hunk 1]
```diff
@@ -311,6 +311,22 @@ func Test_CreateListMembershipForUser_User_Email_Missing(t *testing.T) {
...
 
+func Test_CreateListMembershipForUser_User_Email_Tidepool_Io(t *testing.T) {
+	manager, _ := NewTestManagerWithClientMock(t)
+	newUserMock := NewUserMock()
+	newUserMock.EmailStub = func() string { return "test@tidepool.io" }
+	manager.CreateListMembershipForUser(newUserMock)
+	time.Sleep(time.Second)
+}
+
+func Test_CreateListMembershipForUser_User_Email_Tidepool_Org(t *testing.T) {
```
Comment (jh-bate) [inline user/mailchimp/mailchimp_test.go]: was trying to figure this out but I am missing where you check to see that the `tidepool.org` email isn't added to the list??

Comment (darinkrauss) [reply user/mailchimp/mailchimp_test.go]: It works the same way as the `Test_CreateListMembershipForUser_User_Email_Missing` test. 

Basically, the `clientMock` requires that you specify all outputs to be returned for any method called against the mock. If you don't supply the necessary output and a method is invoked, then the mock will panic (https://github.com/tidepool-org/shoreline/blob/master/user/mailchimp/mailchimp_test.go#L803), thus failing the test. 

In this case, we do NOT specify any outputs and since we don't get a panic, it implies that the `clientMock` was never called to update Mailchimp. So, in effect, the fact that this does NOT panic implies that the code properly ignored the "@tidepool.org" and "@tidepool.io" emails and never tried to call into Mailchimp.

In short, it shows that Mailchimp would NOT have been called for Tidepool email addresses, which is exactly what we are trying to test here.

Yes, it isn't completely obvious, but as with a lot of testing via injection you need to validate the test by inference. If something didn't happen over here then it means that this other thing didn't happen over there.

**Comment by darinkrauss** [comment]
@jh-bate See my response to your question above. Please review and approve.
